controllers/ebpf/agent-metrics.go

```diff
@@ -0,0 +1,90 @@
+package ebpf
+
+import (
+	"context"
+	"k8s.io/apimachinery/pkg/api/errors"
+
+	flowslatest "github.com/netobserv/network-observability-operator/apis/flowcollector/v1beta2"
+	"github.com/netobserv/network-observability-operator/controllers/constants"
+	"github.com/netobserv/network-observability-operator/controllers/reconcilers"
+	"github.com/netobserv/network-observability-operator/pkg/helper"
+	monitoringv1 "github.com/prometheus-operator/prometheus-operator/pkg/apis/monitoring/v1"
+
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/intstr"
+)
+
+func (c *AgentController) ReconcileMetricsService(ctx context.Context, target *flowslatest.FlowCollectorEBPF) error {
+	report := helper.NewChangeReport("EBPF Agent prometheus service")
+	defer report.LogIfNeeded(ctx)
+
+	if err := c.ReconcileService(ctx, c.promSvc, c.promService(target), &report); err != nil && !errors.IsAlreadyExists(err) {
```

Comment:
why `IsAlreadyExists` is checked here? Sounds like not propagating this error could cause missed reconcile events?

Reply:
agree removed the check for alreadyexsists